ap/production/normalization.py

```diff
@@ -39,7 +39,8 @@ def normalization_weights(
     normalization weight.
     """
     # add process ids
-    self.stack.process_ids(events, config_inst=config_inst, dataset_inst=dataset_inst, **kwargs)
+    if "process_id" not in events.fields:
+        self.stack.process_ids(events, config_inst=config_inst, dataset_inst=dataset_inst, **kwargs)
```

Comment:
I think it's okay to always process them. Internally there is a cache that checks of a producer already ran, and if yes, doesn't run it again. it would also catch this one here.

```suggestion
    self.stack.process_ids(events, config_inst=config_inst, dataset_inst=dataset_inst, **kwargs)
```